WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2144,10 +2151,13 @@ public void onMediaRetryClicked(String mediaId) {
 
     @Override
     public void onMediaUploadCancelClicked(String mediaId, boolean delete) {
-        MediaModel media = new MediaModel();
-        media.setMediaId(Long.valueOf(mediaId));
-        MediaPayload payload = new MediaPayload(mSite, media);
-        mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
+        if (!TextUtils.isEmpty(mediaId)) {
+            int localMediaId = Integer.valueOf(mediaId);
+            EventBus.getDefault().post(new PostEvents.PostMediaCanceled(localMediaId));
+        } else {
+            // Passed mediaId is incorrect: cancel all uploads
+            EventBus.getDefault().post(new PostEvents.PostMediaCanceled(true));
```

Comment:
> I understand we are cancelling everything to make sure we also cancel the tapped image as intended by the user, even when cancelling everything else is undesired, as opposed to not cancelling anything.

Not exactly, we're cancelling everything to make sure the upload service state is clean and next uploads will work.